session.go

```diff
@@ -925,27 +925,45 @@ type StmtTxn struct {
 	fail error
 }
 
-func (st *StmtTxn) init() {
+func (st *TxnState) init() {
 	st.buf = kv.NewMemDbBuffer(kv.DefaultTxnMembufCap)
 	st.mutations = make(map[int64]*binlog.TableMutation)
 }
 
 // Valid overrides Transaction interface.
-func (st *StmtTxn) Valid() bool {
+func (st *TxnState) Valid() bool {
 	return st.Transaction != nil && st.Transaction.Valid()
 }
 
-func (st *StmtTxn) changeInvalidToValid(txn kv.Transaction) {
+func (st *TxnState) validEx() bool {
```

Comment:
What does `Ex` mean?

Reply:
I guess it's 'extend'
validEx = valid transaction or future

Reply:
It's not obvious, how about `validOrPending`?
And `changeInalidToFuture` can be renamed to `changeInvalidToPending`?
`changeFutureToValid` can be renamed to `changePendingToValid`.